pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/extension/MetadataExtractorTreeProcessor.java

```diff
@@ -136,18 +137,24 @@ private void extractAbstract(List<StructuralNode> allNodes) {
      */
     private void extractModuleType(Document document) {
         Object attValue = document.getAttribute(MODULE_TYPE_ATT_NAME);
+        ModuleMetadata mm = null;
+        if (documentMetadata instanceof ModuleMetadata) {
```

Comment:
I dislike `instanceof` 😢 
I wonder if extracting the module type is something we should externalize to a different asciidoctor extension. One that specializes in `Module`s. That way we build the extension set based on the type of document that we need to generate.

Reply:
Per our gchat conversation, I feel that this is a strength of the code, not a weakness. I would like to leave it as-is for now. We could revisit the decision later if things get more complicated.